core/src/main/java/org/apache/accumulo/core/data/ArrayByteSequence.java

```diff
@@ -180,4 +180,14 @@ public byte[] toArray() {
   public String toString() {
     return new String(data, offset, length, UTF_8);
   }
+
+  public void setArray(byte[] data, int offset, int len) {
+    this.data = data;
+    this.offset = offset;
+    this.length = len;
+  }
+
+  public void setLength(int len) {
+    this.length = len;
+  }
```

Comment:
In the constructor, there are bounds checks to make sure that the length and offset values make sense. Those same checks should appear here or else there will be exceptions later.

Similarly, the setLength parameter should be checked to make sure it's a valid length.

Reply:
Done, updated.